README.md

```diff
@@ -128,23 +168,72 @@ impl FutureService for HelloServer {
         Ok(format!("Hello, {}!", name))
     }
 }
+```
+
+In this example, `hello` returns `Result`, but in more complex servers, you might talk to another
+server or do something using IO. In those cases, the `HelloFut` type would be something more
+complex. An example of such a server can be found in `examples/two_servers.rs`. In cases where RPCs
+are very fast and don't block, implementing `FutureService` is a good choice, since there's no
+threading overhead. However, blocking in an RPC method implementation would cause the reactor core
+running the server to grind to a halt, which would prevent new RPCs from being served.
+
+```rust
+extern crate future_starter;
+extern crate tarpc;
+extern crate tokio_core;
+
+use future_starter::FutureServiceExt;
+use tarpc::server::Options;
+use tarpc::util::FirstSocketAddr;
 
 fn main() {
-    let mut reactor = reactor::Core::new().unwrap();
-    let (handle, server) = HelloServer.listen("localhost:10000".first_socket_addr(),
-                                  &reactor.handle(),
-                                  server::Options::default())
-                          .unwrap();
-    reactor.handle().spawn(server);
-    let options = client::Options::default().handle(reactor.handle());
-    reactor.run(FutureClient::connect(handle.addr(), options)
+    let mut reactor = tokio_core::reactor::Core::new().unwrap();
+    let (addr, server) = future_starter::HelloServer.listen("localhost:0".first_socket_addr(),
+                &reactor.handle(),
+                Options::default())
+        .unwrap();
+    println!("Listening on {}", addr);
+    reactor.run(server).unwrap();
+}
+```
+
+Here, the `HelloServer.listen("localhost:0", ..)` call binds to an ephemeral port chosen by the
```

Comment:
Same comment as above, using a specific port would make it easier for users to copy-paste the client code without changing anything.